app/models/theme.rb

```diff
@@ -188,25 +201,22 @@ def self.clear_default!
     expire_site_cache!
   end
 
-  def self.transform_ids(ids, extend: true)
-    return [] if ids.nil?
-    get_set_cache "#{extend ? "extended_" : ""}transformed_ids_#{ids.join("_")}" do
-      next [] if ids.blank?
-
-      ids = ids.dup
-      ids.uniq!
-      parent = ids.shift
+  def self.transform_ids(id)
```

Comment:
Drop support for querying for multiple themes. The previous behaviour was too confusing as it assumes the first element in the array is the parent theme id while the other elements in the array are the components theme ids of the parent.

Reply:
I think it might be good to add a validation here to make sure there isn't anywhere still passing an array. I took this PR for a spin locally, and it looks like theme modifiers are broken because of this issue (`ThemeModifierHelper` and `Theme.lookup_modifier` handle arrays).

Easy way to try this out is to install https://meta.discourse.org/t/topic-list-excerpts-theme-component/151520 locally as a component.